cleanlab_studio/studio/enrichment.py

```diff
@@ -184,38 +192,87 @@ def run(
+        return api.get_enrichment_job_status(self._api_key, job_id=latest_job["id"])
+
+    def _get_num_processed_rows(self) -> int:
+        num_rows = int(self._get_enrichment_project_dict()["num_rows"])
+        latest_job_status = self._get_latest_job_status()
+        status = latest_job_status["status"]
+        if status == EnrichmentJobStatusEnum.SUCCEEDED.value:
+            return num_rows
+        elif status == EnrichmentJobStatusEnum.RUNNING.value:
+            return latest_job_status.get("processed_rows") or 0
+        elif status in {
+            EnrichmentJobStatusEnum.FAILED.value,
+            EnrichmentJobStatusEnum.CREATED.value,
+        }:
+            return 0
+        else:
+            raise ValueError("The latest populate job has an unknown status.")
+
     @property
     def ready(self) -> bool:
         """Check if the latest populate job is ready."""
-        latest_job = self.list_all_jobs()[0]
+        latest_job = self._get_latest_job()
         if latest_job["job_type"] != "ENRICHMENT":
             raise ValueError(
                 "The latest job is a preview, to execute against entire dataset, please do `run()` first."
             )
-        self.latest_populate_job = latest_job
-        if latest_job["status"] == "FAILED":
+        status = latest_job["status"]
+        if status == EnrichmentJobStatusEnum.FAILED.value:
             raise ValueError("The latest populate job failed.")
-        elif latest_job["status"] == "RUNNING":
-            return False
-        elif latest_job["status"] == "CREATED":
+        elif status in {
+            EnrichmentJobStatusEnum.RUNNING.value,
+            EnrichmentJobStatusEnum.CREATED.value,
+        }:
             return False
-        elif latest_job["status"] == "SUCCEEDED":
+        elif status == EnrichmentJobStatusEnum.SUCCEEDED.value:
             return True
         else:
             raise ValueError("The latest populate job has an unknown status.")
 
     def wait_until_ready(self) -> None:
         """Wait until the latest populate job is ready."""
-        while not self.ready:
-            time.sleep(CHECK_READY_INTERVAL)
-            pass
+        num_rows = self._get_enrichment_project_dict()["num_rows"]
+        spinner = itertools.cycle("|/-\\")
+        with tqdm(
+            total=num_rows,
+            desc="Enrichment Progress:",
+            bar_format="{desc} Rows Processed - {n_fmt}/{total_fmt}{postfix}",
+        ) as pbar:
+            while not self.ready:
+                latest_job_status = self._get_latest_job_status()
+                num_processed_rows = self._get_num_processed_rows()
+                if pbar.total is None and num_rows is not None:
+                    pbar.total = num_rows
+                    pbar.refresh()
+
+                pbar.set_postfix_str(latest_job_status["status"])
+                pbar.update(num_processed_rows - pbar.n)
+
+                for _ in range(50):
+                    time.sleep(0.1)
```

Comment:
why not just `sleep(50)`? and i think 2mins should be a sweet spot since this is average time for tlm to process data

Reply:
the purpose of this code is create a progress indicator with a spinner animation, rather than just pausing for X seconds. total sleep time here will be 50 * 0.1 = 5 seconds. we'll want to poll more frequently to dynamically get progress percentages, but we can configure the proper polling frequency. i've re-introduced `CHECK_READY_INTERVAL` with a value of 100, so total sleep time in each while loop iteration is now 10 seconds